### PR TITLE
[DQL3-2] - added resource visibility field

### PR DESCRIPTION
**Pull request by awset**
https://salsadigital.atlassian.net/browse/DQL3-2

image

### CODE REVIEW

**Comment by duttonw** [comment]
please fix lint issues:

==> Lint code
/app/ckanext/data_qld/resource_visibility/helpers.py:8: [F401] 'pprint.pformat' imported but unused
from pprint import pformat
^
/app/ckanext/data_qld/resource_visibility/validators.py:3: [F401] 'pprint.pformat' imported but unused
from pprint import pformat
^

**Review by tino097 — APPROVED** [review]
LGTM

File: ckanext/data_qld/resource_visibility/helpers.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import ckan.plugins.toolkit as toolkit
+import logging
+
+from ckan import authz
+from ckan.common import request
```
Comment (MarkCalvert) [inline ckanext/data_qld/resource_visibility/helpers.py]: Use toolkit.request instead

File: ckanext/data_qld/resource_visibility/helpers.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+from ckan.model.package import Package
+from ckan.plugins.toolkit import get_action
```
Comment (MarkCalvert) [inline ckanext/data_qld/resource_visibility/helpers.py]: toolkit has already been imported at the top.
Set the `get_action` variable with the `h` variable below

File: ckanext/data_qld/resources_plugin.py [hunk 3]
```diff
@@ -11,8 +11,11 @@
 import helpers
 import logging
 
+from ckan.common import request
```
Comment (MarkCalvert) [inline ckanext/data_qld/resources_plugin.py]: Use toolkit.request instead

File: ckanext/data_qld/resources_plugin.py [hunk 4]
```diff
@@ -11,8 +11,11 @@
...
 from flask import Blueprint
 from de_identified_data import helpers as de_identified_data_helpers
+from resource_visibility import helpers as resource_visibility_helpers
```
Comment (MarkCalvert) [inline ckanext/data_qld/resources_plugin.py]: Import helpers using full path namespace for CKAN 2.9 compatibility eg. `from ckanext.data_qld.resource_visibility import helpers`
Can you update the `de_identified_data` helper import also

File: ckanext/data_qld/resources_plugin.py [hunk 5]
```diff
@@ -90,6 +98,10 @@ def before_create(self, context, data_dict):
     def before_update(self, context, current_resource, updated_resource):
         return self.check_file_upload(updated_resource)
 
+    def before_show(self, resource_dict):
+        if request.path_info.split('/')[1] == 'api':
```
Comment (MarkCalvert) [inline ckanext/data_qld/resources_plugin.py]: FYI You can also use toolkit.get_endpoint() which returns a tuple (controller|blueprint, action|view).
As far as I am aware any api calls should return `toolkit.get_endpoint()[1] == 'action'`

File: test/features/schema_metadata.feature [hunk 6]
```diff
@@ -33,3 +33,63 @@ Feature: SchemaMetadata
         When I log in
         When I go to "/dataset/warandpeace"
         Then I should see an element with xpath "//link[@type='application/ld+json']"
+
+    Scenario Outline: Scenario 1: A publisher publishes a dataset [DQL3-1]
```
Comment (MarkCalvert) [inline test/features/schema_metadata.feature]: Update all the scenario test titles to match the steps the test is doing.
We do not need use the same title as the Scenario listed as the tests in the story as they are too broad and don't always match the test scenario.
We don't need to reference the story number either

File: test/features/resource_availability.feature [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+    Scenario: A general user views a dataset resource additional info
+        Given "Unauthenticated" as the persona
+
+        And I go to "/dataset/contains-de-identified-data-yes"
+        Then I press the element with xpath "//a[@title='resource created by Admin and is available']"
+        Then I should not see "Re-identification risk governance acknowledgement/Resource visibility"
+
+
+    Scenario: Clean up
```
Comment (MarkCalvert) [inline test/features/resource_availability.feature:202]: Good thinking! Nice way of cleaning up data to re-test locally, never thought about doing that.

**Review by MarkCalvert — CHANGES_REQUESTED** [review]

File: ckanext/data_qld/resource_visibility/helpers.py (hunk 1 above)
Comment (awset) [reply ckanext/data_qld/resource_visibility/helpers.py]: refactored

File: ckanext/data_qld/resource_visibility/helpers.py (hunk 2 above)
Comment (awset) [reply ckanext/data_qld/resource_visibility/helpers.py]: refactored

File: ckanext/data_qld/resources_plugin.py (hunk 3 above)
Comment (awset) [reply ckanext/data_qld/resources_plugin.py]: refactored

File: ckanext/data_qld/resources_plugin.py (hunk 4 above)
Comment (awset) [reply ckanext/data_qld/resources_plugin.py]: refactored

File: ckanext/data_qld/resources_plugin.py (hunk 5 above)
Comment (awset) [reply ckanext/data_qld/resources_plugin.py]: refactored

File: test/features/schema_metadata.feature (hunk 6 above)
Comment (awset) [reply test/features/schema_metadata.feature]: refactored

**Review by MarkCalvert — APPROVED** [review]

**Pull request merged**